server.go

```diff
@@ -74,6 +81,7 @@ type response struct {
 	tsigProvider   TsigProvider
 	udp            net.PacketConn // i/o connection if UDP was used
 	tcp            net.Conn       // i/o connection if TCP was used
+	doq            quic.Stream    // i/o connection if QUIC was used
```

Comment:
can't we check if it's a quick stream when we accept a udp conn? And then do a type assertion on the `net.PacketConn` we have if it implements `Stream()` - or whatever the interface is?

Reply:
A `quic.Stream` is not a `net.PacketConn`, and a `quic.Connection` can have many `quic.Stream`s (each stream will carry one query/response)

Reply:
yes, but https://github.com/lucas-clemente/quic-go/blob/master/example/echo/echo.go#L39 shows it _is_ a conn first and after that some quic stuff happens, I wonder if we could do the same to avoid much code duplication (this may be impossible)